lib/notifications/homeassistant-mqtt.js

```diff
@@ -0,0 +1,54 @@
+import mqtt from "mqtt";
+import { config } from "../config.js";
+
+var mqttConnection = null;
+var entitiesCache = {}
+
+function getMqttConnection() {
+    if (!mqttConnection)
+        mqttConnection = mqtt.connect(config.get('notifications.mqtt.url'), {
+            username: config.get("notifications.mqtt.username"),
+            password: config.get("notifications.mqtt.password")
+        });
+    return mqttConnection;
+}
+
+export function notifyMQTT(messages) {
+    messages.forEach(message => {
+        publishStock(
+            `TooGoodToGo ${message.name}`,
+            `toogoodtogo_${message.id}`,
+            message.quantity > 0,
+            message
+        )
+    });
+}
+
+function publishStock(deviceName, uniqueId, state, extraAttributes = {}) {
```

Comment:
Just take only the message as param and construct the rest inside this function. Spreading the logic like this makes it less clear.